### ISSUE TITLE
Table of contents - not working for asciidoctor-web-pdf (but working for asciidoctor-pdf )

### ISSUE THREAD

**Issue by vishalkrsinha** [Mogztter/asciidoctor-web-pdf#534]
```
:doctype: book
:PubDate: October 2020
:SystemType: Eductor
:KSDocType: Module Reference

= {SystemType}
ifdef::backend-pdf[]
:toc: macro
endif::[]


:sectnums:
:experimental:
:RevNumber: B
:icons: font
:linkattrs:
:webfonts!:
:data-uri:
:linkcss:
:iconfont-remote!:
:webfonts!:
:stylesdir: ../../../../../DocResources/css

<<<

*Document history*

[cols="12,15,73"]
|===
3+|{set:cellbgcolor:#ccc}
|{set:cellbgcolor:}Rev. A
|March 2013
|Created document and first XML version -- MHy
|Rev. B
|October 2020
|Design calculations -- MHy

|===


ifdef::backend-pdf[]
toc::[]
endif::[]



== Introduction
....
```

Above script generates proper toc with 'asciidoctor-pdf' as below...
![toc-asciidoctor-pdf](https://user-images.githubusercontent.com/42475844/126992563-cf0905d8-990e-4c49-9ffc-a507e8f37af6.PNG)

But the same script generates not the decent toc with 'asciidoctor-web-pdf' as below...
![toc-asciidoctor-web-pdf](https://user-images.githubusercontent.com/42475844/126992662-2c0e7264-e54b-4901-b3b5-9f405f6909f0.PNG)

Suggestions to correct this will be much appreciated. Thank you.

**Comment by Mogztter** [comment]
`backend-pdf` is not defined in this converter, see: https://github.com/Mogztter/asciidoctor-web-pdf/issues/129

So both conditional blocks won't work:

```
ifdef::backend-pdf[]
:toc: macro
endif::[]


// ...

ifdef::backend-pdf[]
toc::[]
endif::[]
```

You can use `env-web-pdf` or pass a custom attribute.

**Comment by vishalkrsinha** [comment]
> env-web-pdf

Sorry, but got confused with the suggestion(s) here. I modified my script a bit as below (just commented out the conditional parts):

```
:doctype: book
:PubDate: October 2020
:SystemType: Eductor
:KSDocType: Module Reference

= {SystemType}
// ifdef::env-github[]
:toc: macro
:toclevels: 3
// endif::[]

:sectnums:
:experimental:
:RevNumber: B
:icons: font
:linkattrs:
:webfonts!:
:data-uri:
:linkcss:
:iconfont-remote!:
:webfonts!:

<<<

*Document history*

[cols="12,15,73"]
|===
3+|{set:cellbgcolor:#ccc}
|{set:cellbgcolor:}Rev. A
|March 2013
|Created document and first XML version -- MHy
|Rev. B
|October 2020
|Design calculations -- MHy

|===


// ifdef::env-github[]
// toc::[]                          
// endif::[]

== Introduction
```

And I see the below output - No toc on the left  side (in bookmarks) but see on the right side:
![toc](https://user-images.githubusercontent.com/42475844/127123680-1a6e7324-8fc0-4044-8a2a-afae5cdf7b8a.PNG)

I need toc in Bookmark section too with numbering of contents exactly similar to 'asciidoctor-pdf' results. Suggestions please?

**Comment by Mogztter** [comment]
Oh I see, so the Table Of Contents is generated but missing from the PDF outline right?

Probably something we need to fix to be "on par" with `asciidoctor-pdf`.

**Comment by vishalkrsinha** [comment]
> Oh I see, so the Table Of Contents is generated but missing from the PDF outline right?

 Yes. But I believe toc would have been in bookmarks (left pane are) too along with in main document. I might be missing the trick. Do you have any simplest example please for the same?

> Probably something we need to fix to be "on par" with `asciidoctor-pdf`.

**Comment by vishalkrsinha** [comment]
Any update please?

**Comment by Mogztter** [comment]
> Any update please?

This is an open source project, if you want to implement this feature, you will need to update the logic in here: https://github.com/Mogztter/asciidoctor-web-pdf/blob/main/lib/outline.js

**Comment by ggrossetie** [comment]
Retested with a `doctype: book` document using `:toc: macro` and `toc::[]` placed after the preamble, matching the scenario in this issue. The PDF outline (bookmarks panel) correctly lists every chapter/section as a clickable destination. Closing.
